### PR TITLE
feature/trainer/diego

### PR DESCRIPTION
**Pull request by nanrehd**
Please review the code for Trainer and Backpack.
Some changes were made to fit the use of Pokemon and the integration with PokeHealer.

### CODE REVIEW

File: src/main/java/org/awt05/skills/AbleToStrength.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package org.awt05.skills;
 
 public interface AbleToStrength {
-    default void Strength() {
+    default void strength() {
```
Comment (Snip-77) [inline src/main/java/org/awt05/skills/AbleToStrength.java]: Good code smell!

File: src/main/java/org/awt05/Main.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package org.awt05;
-import org.awt05.dummies.Trainer;
+
```
Comment (Snip-77) [inline src/main/java/org/awt05/Main.java]: Space between imports

File: src/main/java/org/awt05/Main.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
+import org.awt05.pokemons.Gengar;
+import org.awt05.trainer.Trainer;
 import org.awt05.pokehealer.PokeHealer;
```
Comment (Snip-77) [inline src/main/java/org/awt05/Main.java]: @nanrehd Here is a space between imports, I leave to your judgment to leave it this way or not

File: src/main/java/org/awt05/Main.java (hunk 2 above)
Comment (nanrehd) [reply src/main/java/org/awt05/Main.java]: This one's a deleted line, so it isn't actually in the code anymore, hence, there's nothing wrong with it.

File: src/main/java/org/awt05/Main.java (hunk 3 above)
Comment (nanrehd) [reply src/main/java/org/awt05/Main.java]: Same as the previous one

File: src/main/java/org/awt05/Main.java [hunk 4]
```diff
@@ -14,7 +16,22 @@ public static void main(String[] args) {
...
+        List<Trainer> trainers = new ArrayList<>();
+
+        for (int i = 0; i < 3; i++) {
+            Trainer trainer = new Trainer("Trainer");
+
+            PokemonContainer backpack = trainer.getBag();
+            backpack.add(new Pokemon());
+            backpack.add(new Pokemon());
+            backpack.add(new Pokemon());
+            backpack.add(new Pokemon());
```
Comment (ArielYanarico) [inline src/main/java/org/awt05/Main.java]: We cannot be able to instantiate Pokemon, Please make "Pokemon" class abstract

File: src/main/java/org/awt05/PokemonStorage.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ * @author: Carlos Meneses Covarrubias
+ * @date: 03/02/2020
+ * @version: 1.0
+ */
+
+public interface PokemonStorage {
+
+    void addPokemon(Pokemon pokemonIn);
+
+    Pokemon retrievePokemon(int idPokemon);
```
Comment (luqp) [inline src/main/java/org/awt05/PokemonStorage.java]: I think that it is not necessary now, sorry for I said you that push it.

File: src/main/java/org/awt05/trainer/Trainer.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    public Backpack getBag() {
+        return bag;
+    }
+
+    public String getTrainerId() {
+        return trainerId;
+    }
+
+    public void setName(String name) {
```
Comment (ArielYanarico) [inline src/main/java/org/awt05/trainer/Trainer.java]: Should name be settable?

File: src/test/java/org/awt05/trainer/BackpackTest.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+class BackpackTest {
+
+    @Test
+    void whenTrainerGetsPokemonAddsToItsTeam() {
+        //given
+        Pokemon pokemon1 = new Pokemon();
+        Backpack backpack = new Backpack();
+
+        //when
+        backpack.add(pokemon1);
```
Comment (luqp) [inline src/test/java/org/awt05/trainer/BackpackTest.java]: I think that you can add an specific test to **.add()**  method.

File: src/test/java/org/awt05/trainer/TrainerTest.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        //then
+        //assert that both trainer have same names
+        assertNotEquals(trainer.toString(), trainerDifferent.toString());
+        //but different IDs
+        assertNotEquals(trainer.getTrainerId(), trainerDifferent.getTrainerId());
+
+    }
+
+    @Test
+    void GivenASettedName_WhenGettingName_ItReturnsTheSameName() {
```
Comment (luqp) [inline src/test/java/org/awt05/trainer/TrainerTest.java]: If follow our standards the methods should be written in camel case.

File: src/test/java/org/awt05/trainer/TrainerTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+        //when
+        String name2 = trainer.getName();
+
+        //then
+        assertEquals(name, name2);
+    }
+
+    @Test
+    void GivenASettedBag_WhenGettingBag_ItReturnsTheSameBag() {
```
Comment (luqp) [inline src/test/java/org/awt05/trainer/TrainerTest.java]: the same cases.` If follow our standards the methods should be written in camel case.`

File: src/main/java/org/awt05/trainer/Trainer.java (hunk 6 above)
Comment (nanrehd) [reply src/main/java/org/awt05/trainer/Trainer.java]: Not now.

File: src/test/java/org/awt05/trainer/BackpackTest.java (hunk 7 above)
Comment (nanrehd) [reply src/test/java/org/awt05/trainer/BackpackTest.java]: You're right

**Pull request merged**